ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveSubQueryRemoveRule.java

```diff
@@ -199,7 +199,7 @@ private RexNode rewriteScalar(RelMetadataQuery mq, RexSubQuery e, Set<Correlatio
 
       // id is appended since there could be multiple scalar subqueries and FILTER
       // is created using field name
-      String indicator = "alwaysTrue" + e.rel.getId();
+      String indicator = "trueLiteral";
```

Comment:
Please move this to a private static constant variable at the class level, so you do not have to declare multiple times same String value.
In addition, can this name clash with any other column name in the operator?